props/prop_base.h

```diff
@@ -110,6 +110,38 @@ class PropBase : CommandParser, Looper, protected SaberBase {
     return SaberBase::IsOn() || on_pending_;
   }
 
+// Match preon.wav to out.wav
+  virtual void SelectIgnitionPair() {
+    if (!SFX_preon.files_found()) return;
+
+    int preonCount = SFX_preon.files_found();
+    int outCount = SFX_out.files_found();
+    int pairIgnition;
+
+    // If we don't have the same number of preon/out files, then don't bother getting a matched pair.
+    if (preonCount == outCount) {
+        pairIgnition = rand() % preonCount;
+        SFX_preon.Select(pairIgnition);
+        SFX_out.Select(pairIgnition);
+    }
+  }
+
+// Match pstoff.wav to in.wav
+  virtual void SelectRetractionPair() {
+    if (!SFX_pstoff.files_found()) return;
+
+    int pstoffCount = SFX_pstoff.files_found();
+    int inCount = SFX_in.files_found();
+    int pairIgnition;
```

Comment:
This should be combined with line 139.

Reply:
Both cleaned up. Also moved pstoff above Off().